analysis/src/github.rs

```diff
@@ -0,0 +1,172 @@
+//! This module abstracts the communication with GitHub API for a given crate
+
+use anyhow::{anyhow, Result};
+use guppy::graph::PackageMetadata;
+use reqwest::header::{HeaderMap, HeaderValue, AUTHORIZATION, USER_AGENT};
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use url::Url;
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct RepoStats {
+    pub full_name: Option<String>,
```

Comment:
maybe it would be interesting to have a last_updated date field, so that we know not to use old repos that haven't been updated in a long time?

Reply:
it could be calculated using the last commit of the main branch perhaps

Reply:
also further additions that would be useful: open github issues with the tag "security"

Reply:
@xvschneider I am working on this on a separate PR that targets generating activity metrics. I plan to get these info you mentioned in a new PR.

Reply:
great!